rust/routee-compass/src/app/compass/compass_app.rs

```diff
@@ -142,13 +144,18 @@ impl TryFrom<(&Config, &CompassAppBuilder)> for CompassApp {
             .try_deserialize::<serde_json::Value>()?
             .normalize_file_paths(&"", &root_config_path)?;
 
-        let alg_params = config_json.get_config_section(CompassConfigurationField::Algorithm)?;
+        let alg_params =
+            config_json.get_config_section(CompassConfigurationField::Algorithm, &"TOML")?;
         let search_algorithm = SearchAlgorithm::try_from(&alg_params)?;
 
+        let state_params =
```

Comment:
I wonder if we could nest the state parameters within a toml section? I think this setup would require us to put the `state` key at the top of the toml file since if it's not then it will get nested inside another section. For example if we have the config:

```toml
[graph]
edge_list_input_file = "../data/tomtom_denver/edges-compass.csv.gz"
vertex_list_input_file = "../data/tomtom_denver/vertices-compass.csv.gz"
verbose = true

state = [
    { distance_unit = "kilometers", initial = 0.0 },
    { time_unit = "minutes", initial = 0.0 },
]
```

state would get nested under the `graph` key. Maybe the config could look something like:

```toml
[state_model]
state = [
    { distance_unit = "kilometers", initial = 0.0 },
    { time_unit = "minutes", initial = 0.0 },
]
```

Reply:
ya, you're right, and i thought this was awkward too. i like your suggestion and i think it's a simple change.